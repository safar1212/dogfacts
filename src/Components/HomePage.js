// import KeyboardVoiceIcon from '@mui/icons-material/KeyboardVoice';
import React from "react";

const HomePage = () => {
    return (
        <div>
         <h1>Test homepage This is a try</h1>
        </div>
        
    )
}

export default HomePage;