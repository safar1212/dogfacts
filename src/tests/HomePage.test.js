import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import axios from 'axios';
import store from '../redux/configureStore';
import HomePage from '../Components/HomePage';
// import { dogActions, getDogs } from '../Redux/mainPage';

jest.mock('axios');

describe('Details test', () => {
  it('Details render correctly', () => {
    const DetailsPage = renderer.create(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    ).toJSON();
    expect(DetailsPage).toMatchSnapshot();
  });

  test('Render Details component', () => {
    render(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    );
  });

  const response = [
    {
      name: 'terrior',
      id: 0,
    },
    {
      name: 'Austrialian',
      id: 1,
    },
    {
      name: 'WesternSahara',
      id: 2,
    },
    {
      name: 'Zambian bull',
      id: 3,
    },
    {
      name: 'asian hound',
      id: 4,
    },
  ];

  test('integration testing', async () => {
    axios.get.mockResolvedValue({ data: response });

    render(
      <Provider store={store}>
        <HomePage key={response.id} coin={response} />
      </Provider>,
    );

    const userList = await waitFor(() => screen.getByTestId('user-list'));
    expect(userList).toBeInTheDocument();
  });

  test(' second integration testing', () => {
    axios.get.mockResolvedValue({ data: response });

    render(
      <Provider store={store}>
        <HomePage key={response.id} coin={response} />
      </Provider>, { wrapper: Router },
    );

    const userList = (() => screen.findAllByTestId('user-item'));
    expect(userList).toHaveLength(0);
  });
});
