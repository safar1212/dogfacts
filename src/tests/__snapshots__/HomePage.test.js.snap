// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test Details render correctly 1`] = `
<div
  className="page"
>
  <div
    className="main-section"
  >
    <img
      alt="dog-pic"
      src="main-pic.png"
    />
    <div
      className="main-count"
    >
      <h2>
        Total Breeds
      </h2>
      <h2>
        0
      </h2>
    </div>
  </div>
  <div
    className="sort-wrap"
  >
    <div
      className="input-div"
    >
      <span>
        Search for breed:
      </span>
      <input
        onChange={[Function]}
        placeholder="breed..."
        type="text"
      />
    </div>
    <div
      className="sort-dogs"
    >
      <span
        className="sort-title"
      >
        Sort By Order:
      </span>
      <select
        className="sort"
        onChange={[Function]}
      >
        <option
          value="sort"
        >
          Sort
        </option>
        <option
          value="a-z"
        >
          A to Z
        </option>
        <option
          value="z-a"
        >
          Z to A
        </option>
      </select>
    </div>
  </div>
</div>
`;
